viz_scripts/scaffolding.py

```diff
@@ -105,46 +105,144 @@ def data_quality_check(expanded_ct):
 
 def unit_conversions(df):
     df['distance_miles']= df["distance"]*0.00062 #meters to miles
+    df['duration_h'] = df['duration'] / 60 / 60 #seconds to hours
+
+
+def feat_eng(data, const, feats, prefs, mode='Mode_confirm', repm='Replaced_mode'):
+    """
+    Calculate trip aggregate results from constants and append to data (Feature Engine)
+
+    Parameters:
```

Comment:
Ah, I see that you have refactored more code 😄 
I will have to look at it in detail but glad to see that you are thinking about this.

Reply:
This is actually the generalized function for calculating energy_intensity, cost, and time for each trip. I have yet to generalize the impact calculation function.

Reply:
Yup. You are definitely on the right track with this change. I am actually OK with leaving `*_impact` functions unchanged - the number of duplicated lines is very small. But of course, unifying them would be better...